TDX: add dbs-tdx crate

studychao opened the pull request:
We will first add tdx-ioctls for dbs-tdx.

Signed-off-by: fengshifang <fengshifang@linux.alibaba.com>
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
TDX: add dbs-tdx crate

We will first add tdx-ioctls for dbs-tdx.

Signed-off-by: fengshifang <fengshifang@linux.alibaba.com>
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
Merge branch 'main' into chao/add_dbs_tdx

wllenyj commented on crates/dbs-tdx/src/tdx_ioctls.rs:
```diff
@@ -0,0 +1,221 @@
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+#![allow(missing_docs)]
```
remove?

wllenyj commented on crates/dbs-tdx/src/tdx_ioctls.rs:
```diff
@@ -0,0 +1,221 @@
+}
+
+/// Initialize TDX vcpu
+pub fn tdx_init_vcpu(vcpu_fd: &RawFd, hob_address: u64) -> std::result::Result<(), TdxIoctlError> {
```
use std::result::Result;

studychao replied:
fixed.

NK-fengsf commented on crates/dbs-tdx/src/tdx_ioctls.rs:
```diff
@@ -0,0 +1,221 @@
+#[repr(C)]
+#[derive(Default)]
+/// Tdx capabilities.
+pub struct TdxCapabilities {
```
Add documention for  struct TdxCapabilities

studychao replied:
added.

studychao committed:
Merge branch 'main' into chao/add_dbs_tdx

NK-fengsf commented:
This crate may be used  temporarily，the related function may be implemented in rust-vmm/kvm-ioctls after tdx is supported in upstream  guest kernel.

studychao committed:
Merge branch 'main' into chao/add_dbs_tdx

studychao committed:
tdx: remove allow missing_doc

Since we add document to all pub elements, we could remove allow missing
doc now.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>
